Source/Krypton Components/Krypton.Workspace/Controls Workspace/KryptonWorkspace.cs

```diff
@@ -2899,7 +2899,7 @@ internal void InternalPageDragStart(object sender, KryptonNavigator? navigator,
                     }
 
                     // Calling DragStart will cause the drag targets to be created from the target providers
-                    e.Cancel = !_dragManager.DragStart(e.ScreenPoint, new PageDragEndData(sender as KryptonNavigator, e.Pages));
+                    e.Cancel = !_dragManager.DragStart(e.ScreenPoint, new PageDragEndData((sender as KryptonNavigator)!, e.Pages));
```

Comment:
Either 
- check the nullability from the `as` before sending
- DragStart to accept null's
- Use a cast.

Option 2 preferred.

Reply:
@Smurf-IV 
Dragstart already does a null check on the dragEndData param and will throw if null.
The PageDragEndData class expects a valid source param.

This way the sender is checked and dragstart handles the throw
```cs
PageDragEndData? dragEndData = sender is null
    ? null
    : new PageDragEndData((sender as KryptonNavigator)!, e.Pages);

e.Cancel = !_dragManager.DragStart(e.ScreenPoint, dragEndData);
```

Your thoughts please ?

Reply:
The `sender is null` should have the following check in it `(sender as KryptonNavigator)`
ReSharper will then change it to a null check pattern along the lines of `(sender is KryptonNavigator kn)` and pass the `kn` into the constructor.